src/services/action/get-actions-task.ts

```diff
@@ -0,0 +1,70 @@
+// global
+import {
+  Actor,
+  DatabaseTransactionHandler,
+  UnknownExtra,
+} from 'graasp';
+// local
+import { ActionService } from '../../db-service';
+import { BaseActionTask } from './base-action-task';
+import { Action } from '../../interfaces/action';
+import { views, DEFAULT_ACTIONS_SAMPLE_SIZE, MIN_ACTIONS_SAMPLE_SIZE, MAX_ACTIONS_SAMPLE_SIZE } from '../../constants/constants';
+
+
+export class GetActionsTask<E extends UnknownExtra> extends BaseActionTask<Action[]> {
+
+  readonly itemId: string;
+  readonly requestedSampleSize: string | undefined;
+  readonly view: string | undefined;
+
+
+  get name(): string {
+    return GetActionsTask.name;
+  }
+
+  constructor(actor: Actor, itemId: string, requestedSampleSize: string, view: string, actionService: ActionService) {
+    super(actor, actionService);
+    this.itemId = itemId;
+    this.requestedSampleSize = requestedSampleSize;
+    this.view = view;
```

Comment:
Could `View` be an enum? Then it will be usable as a type.

Reply:
I have created the enum View with the possible view names. Now, the view variable is of type View in all the parts of the project.

Reply:
I have also changed to `enum` `ACTION_TYPES` and `METHODS`